.github/workflows/report.yml

```diff
@@ -95,77 +148,134 @@ jobs:
+      - run: make test
+        env:
+          ACTION_TYPE: CI
+          RUBY_VERSION: ${{ inputs.ruby_version }}
+          NODE_COUNT: ${{ matrix.node_count }}
+          NODE_INDEX: ${{ matrix.node_index }}
+          JUNIT_REPORT_DIR_OLD: old-rspec-reports
+          JUNIT_REPORT_DIR_NEW: rspec-reports
+          GITHUB_TOKEN: ${{ secrets.github_pat }}
+          COVERAGE: ${{ inputs.use_simplecov_report }}
+          ALL_CHANGED_FILES: ${{ needs.dependency-resolution.outputs.all_changed_files }}
+      - name: upload coverage report
+        if: ${{ inputs.use_simplecov_report == true }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage-report-${{ github.run_id }}
+          path: ${{ inputs.simplecov_report_path }}
+          if-no-files-found: warn
+      - name: upload rspec report
         uses: actions/upload-artifact@v2
         with:
           name: rspec-reports
           path: rspec-reports
-          if-no-files-found: error
+          if-no-files-found: warn
 
   store-simplecov-report:
-    name: merge and store coverage
-    needs: [test]
-    if: ${{ success() && needs.test.outputs.allow_simplecov_action == 'true' }}
+    if: ${{ inputs.use_simplecov_report == true && (always() && contains(needs.*.result, 'success') && !(contains(needs.*.result, 'failure'))) }}
+    name: store simplecov report
+    needs: [
+      test,
+      test-with-mysql,
+      test-in-parallel,
+      test-in-parallel-with-mysql,
+    ]
     runs-on: ubuntu-${{ startsWith(inputs.mysql_version, '5') && '18.04' || 'latest' }}
     steps:
       - uses: actions/checkout@v2
       - uses: pinnacles/common-cicd-actions/.github/actions/setup-lang@v0.1.0
         with:
-          use_ruby: ${{ inputs.use_ruby }}
+          use_ruby: 'true'
           ruby_version: ${{ inputs.ruby_version }}
-          use_node: ${{ inputs.use_node }}
-          node_version: ${{ inputs.node_version }}
-          node_package_manager: ${{ inputs.node_package_manager }}
-          use_go: ${{ inputs.use_go }}
-          go_version: ${{ inputs.go_version }}
       - uses: actions/download-artifact@v2
         with:
-          name: tmp-coverage-report
+          name: coverage-report-${{ github.run_id }}
           path: coverage
-      - run: bundle exec rake coverage:report
+        continue-on-error: true
+      - run: ${{ inputs.simplecov_command }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: coverage-head
+          path: |
+            coverage/assets/
+            coverage/coverage.json
+            coverage/index.html
+          if-no-files-found: warn
       - uses: actions/upload-artifact@v2
         with:
-          name: coverage
+          name: coverage-${{ github.sha }}
           path: |
             coverage/assets/
             coverage/coverage.json
             coverage/index.html
-          if-no-files-found: error
-      - name: delete tmp-coverage-report
+          if-no-files-found: warn
```

Comment:
upload two artifacts named coverage-head and coverage-${{ github.sha }}.
both of artifact will be downloaded in ci.yml.
first, download coverage-head, and second, download coverage-${{ github.sha }} if it was found in ci.yml.